tools/test-framework/src/chain/chain_binary.rs

```diff
@@ -0,0 +1,68 @@
+use ibc::core::ics24_host::identifier::ChainId;
+use ibc_relayer::config::AddressType;
+
+use crate::util::random::{random_u32, random_unused_tcp_port};
+
+const COSMOS_HD_PATH: &str = "m/44'/118'/0'/0/0";
+const EVMOS_HD_PATH: &str = "m/44'/60'/0'/0/0";
+
+#[derive(Clone, Debug)]
+pub enum ChainBinary {
+    CosmosChain,
+    EvmosChain,
+}
+
+impl ChainBinary {
+    pub fn get_hd_path(&self) -> &str {
+        match self {
+            Self::CosmosChain => COSMOS_HD_PATH,
+            Self::EvmosChain => EVMOS_HD_PATH,
+        }
+    }
+
+    pub fn get_chain_id(&self, prefix: &str, use_random_id: bool) -> ChainId {
+        match self {
+            Self::CosmosChain => {
+                if use_random_id {
+                    ChainId::from_string(&format!("ibc-{}-{:x}", prefix, random_u32()))
+                } else {
+                    ChainId::from_string(&format!("ibc-{}", prefix))
+                }
+            }
+            Self::EvmosChain => ChainId::from_string(&format!("evmos_9000-{}", prefix)),
+        }
+    }
+
+    pub fn get_json_rpc_address(&self) -> (String, String) {
```

Comment:
Let's have this return an option instead, and perhaps call it `extra_start_args` or something similar?

```suggestion
    pub fn extra_start_args(&self) -> Option<String> {
```

Reply:
Having a single `String` for the flag and argument trigger the following error `Error: unknown flag: --json-rpc.address localhost:50613`.
I will go with the solution by renaming to `extra_start_args` as this would still be clear if there are other flags which are required by other chains.